packages/miniflare/src/workers/queues/broker.worker.ts

```diff
@@ -320,28 +340,36 @@ export class QueueBrokerObject extends MiniflareDurableObject<QueueBrokerObjectE
 		this.#pendingFlush = { immediate: delay === 0, timeout };
 	}
 
-	#enqueue(messages: QueueIncomingMessage[]) {
+	#enqueue(messages: QueueIncomingMessage[], globalDelay: number = 0) {
 		for (const message of messages) {
 			const randomness = crypto.getRandomValues(new Uint8Array(16));
 			const id = message.id ?? Buffer.from(randomness).toString("hex");
 			const timestamp = new Date(message.timestamp ?? this.timers.now());
 			const body = deserialise(message);
-			this.#messages.push(new QueueMessage(id, timestamp, body));
+			const msg = new QueueMessage(id, timestamp, body);
+
+			const fn = () => {
+				this.#messages.push(msg);
+				this.#ensurePendingFlush();
+			}
+
+			const delay = message.delaySecs ?? globalDelay;
+			this.timers.setTimeout(fn, delay * 1000);
 		}
-		this.#ensurePendingFlush();
 	}
 
 	@POST("/message")
 	message: RouteHandler = async (req) => {
 		validateMessageSize(req.headers);
 		const contentType = validateContentType(req.headers);
+		const delay = validateMessageDelay(req.headers);
 		const body = Buffer.from(await req.arrayBuffer());
 
 		// If we don't have a consumer, drop the message
 		const consumer = this.#maybeConsumer;
 		if (consumer === undefined) return new Response();
 
-		this.#enqueue([{ contentType, body }]);
+		this.#enqueue([{ contentType, delaySecs: undefined, body }], delay);
```

Comment:
Why does this set the `globalDelay` instead of the message delay?

Reply:
This is the same in practice; i.e. only one delay can be specified when sending a single message.

Reply:
Coudl you add a comment here to that effect?